plugins/cilium-docker/driver/driver.go

```diff
@@ -417,28 +430,11 @@ func (driver *driver) joinEndpoint(w http.ResponseWriter, r *http.Request) {
 	}
 
 	log.WithField(logfields.Object, old).Debug("Existing endpoint")
-	if old.Status.State != models.EndpointStateCreating {
-		sendError(w, "Error: endpoint not in creating state", http.StatusBadRequest)
-		return
-	}
```

Comment:
Correct. But that was brittle anyway.

Reply:
For the record, I had to remove this check, because now an endpoint can be in any state shortly after creation: waiting-for-identity, waiting-to-regenerate, regenerating, or ready. Therefore there was no point trying to assert which state the endpoint is.